docs/extending/README.md

```diff
@@ -0,0 +1,61 @@
+# Extensibility Overview
+
+## Using APIs
+
+Visual Studio Code Kubernetes Tools offers APIs for integrating with the extension.
+Currently the only API available is for writing cluster providers.  If you would like an
+API for a feature which isn't currently surfaced, please post an issue on GitHub!
+
+### Getting an API object
+
+To access an API, you request an interface for the component you want to access, specifying
+a particular version of the component API. Versioned interfaces are immutable, and existing
+versions should continue to work even after a new version is available, though old versions
+may eventually be retired.
+
+The easiest way to get an interface is to use the [vscode-kubernetes-tools-api NPM package](https://link/tba).
```

Comment:
im sure you're aware, but this package isn't published yet

Reply:
Good point.  Added `do-not-merge` label back in!